app/tests/cases_tests/test_permissions.py

```diff
@@ -54,36 +55,38 @@ def test_permissions(self, user, access):
 
 
 @pytest.mark.django_db
-def test_image_permission_with_algorithm_result():
+def test_image_permission_with_public_job():
     g_reg_anon = Group.objects.get(
         name=settings.REGISTERED_AND_ANON_USERS_GROUP_NAME
     )
     g_reg = Group.objects.get(name=settings.REGISTERED_USERS_GROUP_NAME)
 
-    result = AlgorithmResultFactory()
-    result_image = ImageFactory()
-    result.images.add(result_image)
+    job = AlgorithmJobFactory()
 
-    assert "view_image" not in get_perms(g_reg, result_image)
-    assert "view_image" not in get_perms(g_reg_anon, result_image)
-    assert "view_image" not in get_perms(g_reg, result.job.image)
-    assert "view_image" not in get_perms(g_reg_anon, result.job.image)
+    output_image = ImageFactory()
+    civ = ComponentInterfaceValueFactory(image=output_image)
+    job.outputs.add(civ)
 
-    result.public = True
-    result.save()
+    assert "view_image" not in get_perms(g_reg, output_image)
+    assert "view_image" not in get_perms(g_reg_anon, output_image)
+    assert "view_image" not in get_perms(g_reg, job.inputs.first().image)
+    assert "view_image" not in get_perms(g_reg_anon, job.inputs.first().image)
 
-    assert "view_image" not in get_perms(g_reg, result_image)
-    assert "view_image" in get_perms(g_reg_anon, result_image)
-    assert "view_image" not in get_perms(g_reg, result.job.image)
-    assert "view_image" in get_perms(g_reg_anon, result.job.image)
+    job.public = True
+    job.save()
 
-    result.public = False
-    result.save()
+    assert "view_image" not in get_perms(g_reg, output_image)
+    assert "view_image" in get_perms(g_reg_anon, output_image)
+    assert "view_image" not in get_perms(g_reg, job.inputs.first().image)
+    assert "view_image" in get_perms(g_reg_anon, job.inputs.first().image)
```

Comment:
just for my own understanding, making a job public doesn't mean everyone can view the images? Only the users in the registered and anonymous group?

Reply:
Yes - reg_anon is everyone who is registered plus anonymous users. This was requested as people want to show results without having to sign up.